### DISCUSSION TITLE
How do I write a plugin or external app to interface with the Copilot Chat window?

### DISCUSSION THREAD

**Discussion by otac0011** [microsoft/vscode-discussions#1031] [Extension Development QnA] [answered]
I want to write an extension that programmatically puts text into the Github Copilot Chat box in the extension. is it possible to identify the chatbox in VS Code?

I tried pywinauto, but I couldn't get it to work.

**Comment by isidorn** [comment, marked as answer]
There is the agent API that is still in proposed. Soon we will have docs to share

https://code.visualstudio.com/blogs/2023/11/13/vscode-copilot-smarter#_extensibility
https://github.com/microsoft/vscode-extension-samples/tree/main/chat-agent-sample
